Lesson 10

ilya-burinskiy opened the pull request.

Commit:
Using Devise

Commit:
Rails Mailer

Commit:
Admin Test resources

Commit:
Admin Answer and Question resources

Commit:
Removed unnecessary sessions/registrations files

Commit:
Setting current_user as test author in Test#create

Commit:
Redirecting admins to admin_tests_path

Commit:
Seeds updated

Commit:
Restored registration_controller

Commit:
Greting flash after sign up

VidgarVii commented on app/controllers/application_controller.rb:
```diff
@@ -1,22 +1,5 @@
-
-  def logged_in?
-    current_user.present?
+  def after_sign_in_path_for(user)
```
у тебя полиморфизм метода замещение. Этим сломал нужный код данного метода. Например запись в сессию локации, куда следует вернуть пользователя.  Тут нужен метод дополнения. Вот рекомендация 
https://www.rubydoc.info/github/plataformatec/devise/Devise/Controllers/Helpers:after_sign_in_path_for

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+
+  def create
+    @test = Test.new(test_params)
+    @test.author = current_usert
```
баг

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+  end
+
+  def test_params
+    params.require(:test).permit(:title, :level)
```
баг

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+    @test = Test.new(test_params)
+    @test.author = current_usert
+    if @test.save
+      redirect_to test_path(@test)
```
баг. кривой копипаст

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+
+  def update
+    if @test.update(test_params)
+      redirect_to test_path(@test)
```
баг

VidgarVii commented on app/views/admin/tests/show.html.erb:
```diff
@@ -0,0 +1,15 @@
+</p>
+
+<%= link_to 'Edit', edit_admin_test_path(@test) %>
+<%= link_to 'Back', admin_tests_path %>
```
как создать вопрос?

VidgarVii commented on app/views/admin/questions/edit.html.erb:
```diff
@@ -1,8 +1,8 @@
   <p><%= form.submit %></p>
 <% end %>
-<%= link_to 'Back', :back %>
+<%= link_to 'Back', admin_test_path(@question.test) %>
```
лучше бы оставил :back

VidgarVii commented on app/views/admin/questions/new.html.erb:
```diff
@@ -1,8 +1,8 @@
   <p><%= form.submit %></p>
 <% end %>
-<%= link_to 'Back', :back %>
+<%= link_to 'Back', admin_test_path(@question.test) %>
```
баг. Какой мб тест у еще не созданного вопроса?

Commit:
ApplicationController#after_sign_in_path_for fix

- Added authenticate_user! callback
- super call in after_sign_in_path_for if current_user is not admin

Commit:
AnswerController fix

- Fixed redirect_to paths
- @answer is conneced to the @question

Commit:
QuestionController redirect_to paths fixed

Commit:
TestController fix

- setting current_user as @test.author
- redirect_to paths are fixed
- category_id added to required test params

Commit:
Answer#validate_max_answers_count_for_question fix

- using question body instead of titel

Commit:
Test show view update

- Added new question link

Commit:
Back links fix

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+    @tests = Test.all
+  end
+
+  def show
```
def show; end

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+    redirect_to admin_tests_path
+  end
+
+  def edit
```
def edit; end

VidgarVii commented on app/controllers/admin/tests_controller.rb:
```diff
@@ -0,0 +1,56 @@
+  def create
+    @test = Test.new(test_params)
+    @test.author = current_user
+    if @test.save
```
не хватает пустой строки м/у условием и объявлением теста

Commit:
Formatting
